resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/filesystem/CreateDirectoryTest.java

```diff
@@ -146,13 +147,10 @@ public void testTargetIsFileInLocalFile() {
 	}
 
 	public void testParentDeviceNotExistsInLocalFile() {
-		if (!Platform.getOS().equals(Platform.OS_WIN32)) {
-			return;
-		}
+		Assume.assumeTrue(Platform.getOS().equals(Platform.OS_WIN32));
```

Comment:
That's a NO-GO as it would permanently add failing test on other platforms instead of "skipping" the testcode.

Reply:
The tests fail because Tycho can't deal with `Assume` the way it should and it fails the tests instead of skipping them. 

I only noticed it after doing this PR (my bad) and I just found out about this bug (https://github.com/eclipse-tycho/tycho/issues/630).

I'm going to have to drop this PR since skipping tests using `Assume` is not doable at the moment :(

Reply:
did you try in the @before?

Reply:
hm, interesting, `@Before` will be ignored in this test class, but if I break free from JUnit3 by *not* (indirectly) extending from  `junit.framework.TestCase` then 2 things happen:

1. `@Before` will be called instead of `setup()`
2. `Assume` will skip tests instead of failing them, even when used inside test methods... which is exactly what I wanted in the first place!

I see there are several test classes that indirectly extend from `junit.framework.TestCase`, breaking this chain and moving to JUnit4 is not a trivial task so I will abort this PR and evaluate a better option (maybe migrating some of the test classes in the hierarchy one by one).

Thank you anyway for your comment, at least now I know the problem is solvable by "upgrading" to JUnit4 👍

Reply:
cheers.